src/java/org/apache/cassandra/db/compaction/SharedCompactionObserver.java

```diff
@@ -34,16 +39,25 @@
 /// if assertions are enabled.
 public class SharedCompactionObserver implements CompactionObserver
 {
+    private static final Logger logger = LoggerFactory.getLogger(SharedCompactionObserver.class);
+
     private final AtomicInteger toReportOnComplete = new AtomicInteger(0);
     private final AtomicReference<Throwable> onCompleteException = new AtomicReference(null);
     private final AtomicReference<CompactionProgress> inProgressReported = new AtomicReference<>(null);
     private final CompactionObserver observer;
 
+    private final CopyOnWriteArrayList<CompactionObserver> compObservers = new CopyOnWriteArrayList<>();
+
     public SharedCompactionObserver(CompactionObserver observer)
     {
         this.observer = observer;
     }
 
+    public void registerObserver(CompactionObserver observer)
```

Comment:
This needs some comments on what happens if it is called after the compactions have started/ended.

Maybe even a check/assert that it is not called after the first `onInProgress`.